### ISSUE TITLE
Fix `selector-pseudo-class-no-unknown` false negatives for non-standard `:slotted`

### ISSUE THREAD

**Issue by Mouvedia** [stylelint/stylelint#6878]
### What steps are needed to reproduce the bug?

```css
:slotted(.foo) {}
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
   "rules": {
     "selector-pseudo-class-no-unknown": true
   }
}
```

### How did you run Stylelint?

https://stylelint.io/demo/

### What did you expect to happen?

Unexpected unknown pseudo-class selector ":slotted"

### What actually happened?

no error

### Proposal to fix the bug

Distinguish between the pseudo-class—which is vue specific—and the standard pseudo-element.

ref:
- pseudo-class: https://vuejs.org/api/sfc-css-features.html#slotted-selectors
- pseudo-element: https://drafts.csswg.org/css-scoping/#slotted-pseudo

**Comment by Mouvedia** [comment]
> To confirm, do you mean the [::slotted()](https://developer.mozilla.org/en-US/docs/Web/CSS/::slotted) pseudo-element, not pseudo-class?

@ybiquitous Ill quote myself:

> Distinguish between the pseudo-class—which is vue specific—and the standard pseudo-element.
>
> ref:
>
>- pseudo-class: [vuejs.org/api/sfc-css-features.html#slotted-selectors](https://vuejs.org/api/sfc-css-features.html#slotted-selectors)
>- pseudo-element: [drafts.csswg.org/css-scoping/#slotted-pseudo](https://drafts.csswg.org/css-scoping/#slotted-pseudo)

So no, I want an errorfor `:slotted()` when `{ "ignorePseudoClasses": [ "slotted"] }` is missing  for the rule `selector-pseudo-class-no-unknown` because it's not standard.
e.g. if you switch from vue to react the unknown pseudo-class wouldn't be detected

> If so, we could see the [selector-pseudo-element-no-unknown](https://stylelint.io/user-guide/rules/selector-pseudo-element-no-unknown) rule false negative with [this demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcD4GcA2EAuGZQBQDoAzCCASgAJgBfEAGhEIEtUYA5AQwFs5EYAPLgAcW+MMmR1wEAHZMA5ghDAAOtPLllIAE4BXFsk3xKq9es3IYLMBghaAtIIs6oEO5ZjdpGO9Nc7pANa+AO7ShuQYujAm5FSqNPSQsoxyAGK2nOwYigBWyDKSsI6KKmoaIMgYAJ4sqIxe4ZoAjACs+ABs+ABMmvEgVEA).

`::slotted()` is not unknown.
i.e. in a non-vue project you want to detect that you are not using `:slotted()` but still allow `::slotted()`

**Comment by ybiquitous** [comment]
@Mouvedia Thanks for the explanation, and sorry for my misunderstanding. 🙇🏼 

I understand your motivation. You mean [`selector-pseudo-class-no-unknown`](https://stylelint.io/user-guide/rules/selector-pseudo-class-no-unknown/) should error for the non-standard `:slotted` pseudo-class. This bug is a false negative of `selector-pseudo-class-no-unknown`.

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.
